core-connector/src/main/java/com/modusbox/client/validator/GetPartyResponseValidator.java

```diff
@@ -0,0 +1,37 @@
+package com.modusbox.client.validator;
+
+import com.modusbox.client.customexception.CCCustomException;
+import com.modusbox.client.customexception.CloseWrittenOffAccountException;
+import com.modusbox.client.enums.ErrorCode;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.json.JSONObject;
+
+import java.util.Map;
+
+public class GetPartyResponseValidator implements Processor {
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        String body = exchange.getIn().getBody(String.class);
+        JSONObject respObject = new JSONObject(body);
+
+        if (respObject.has("status")) {
+            String status = respObject.getString("status");
+            if (!status.equals("SUCCESS")) {
+                String errorCode = respObject.getJSONObject("result").getString("errorCode");
+                String errorReason = respObject.getJSONObject("result").getString("errorReason");
+
+               if(errorCode.equals("FBF006")) {
+                   throw new CloseWrittenOffAccountException(errorReason);
```

Comment:
There is no need to throw an error here, you can simply set the exchange body to whatever message you want. This way you don't have to go through error handler to catch the error later and change status code.